go/build.sh

```diff
@@ -9,4 +9,5 @@ popd || exit
 # Build sample app
 
 cd ../opentelemetry-lambda/go/sample-apps/function || exit
+go get github.com/golang/protobuf@latest
```

Comment:
Why is this necessary now?

Reply:
Doing this resolves the[ protobuf failure](https://github.com/aws-observability/aws-otel-lambda/actions/runs/9020406554/job/24785526439#step:12:57) in workflow - https://github.com/protocolbuffers/protobuf/issues/16163,

Reply:
updated the PR to include sub module changes, instead of modifying the build script.